altair/datasets/_typing.py

```diff
@@ -0,0 +1,251 @@
+        Remote url used to access dataset.
+
+    .. _Path.stem:
+        https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem
+    .. _Path.name:
+        https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name
+    .. _Path.suffix:
+        https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix
+    .. _vega-datasets release:
+        https://github.com/vega/vega-datasets/releases
+
+    Examples
+    --------
+    ``Metadata`` keywords form constraints to filter a table like the below sample:
+
+    ```
+    shape: (2_879, 9)
+    ┌───────────┬──────────┬──────────┬──────────┬───┬────────┬─────────┬──────────┐
+    │ dataset_n ┆ ext_supp ┆ file_nam ┆ name_col ┆ … ┆ suffix ┆ tag     ┆ url_npm  │
+    │ a…        ┆ or…      ┆ e        ┆ li…      ┆   ┆ ---    ┆ ---     ┆ ---      │
+    │ ---       ┆ ---      ┆ ---      ┆ ---      ┆   ┆ str    ┆ enum    ┆ str      │
+    │ str       ┆ bool     ┆ str      ┆ bool     ┆   ┆        ┆         ┆          │
+    ╞═══════════╪══════════╪══════════╪══════════╪═══╪════════╪═════════╪══════════╡
+    │ cars      ┆ true     ┆ cars.jso ┆ false    ┆ … ┆ .json  ┆ v1.21.0 ┆ https:// │
+    │           ┆          ┆ n        ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ flights-2 ┆ true     ┆ flights- ┆ true     ┆ … ┆ .arrow ┆ v1.31.1 ┆ https:// │
+    │ 0…        ┆          ┆ 20…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ flights-2 ┆ true     ┆ flights- ┆ false    ┆ … ┆ .json  ┆ v2.9.0  ┆ https:// │
+    │ 0…        ┆          ┆ 20…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ unemploym ┆ true     ┆ unemploy ┆ false    ┆ … ┆ .json  ┆ v2.7.0  ┆ https:// │
+    │ e…        ┆          ┆ me…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ ffox      ┆ false    ┆ ffox.png ┆ false    ┆ … ┆ .png   ┆ v2.5.2  ┆ https:// │
+    │           ┆          ┆          ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ …         ┆ …        ┆ …        ┆ …        ┆ … ┆ …      ┆ …       ┆ …        │
+    │ flights-a ┆ true     ┆ flights- ┆ false    ┆ … ┆ .csv   ┆ v1.18.0 ┆ https:// │
+    │ i…        ┆          ┆ ai…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ income    ┆ true     ┆ income.j ┆ false    ┆ … ┆ .json  ┆ v1.21.0 ┆ https:// │
+    │           ┆          ┆ so…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ burtin    ┆ true     ┆ burtin.j ┆ false    ┆ … ┆ .json  ┆ v2.8.0  ┆ https:// │
+    │           ┆          ┆ so…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ flights-5 ┆ true     ┆ flights- ┆ false    ┆ … ┆ .json  ┆ v1.8.0  ┆ https:// │
+    │ k         ┆          ┆ 5k…      ┆          ┆   ┆        ┆         ┆ cd…      │
+    │ wheat     ┆ true     ┆ wheat.js ┆ false    ┆ … ┆ .json  ┆ v1.18.0 ┆ https:// │
+    │           ┆          ┆ on       ┆          ┆   ┆        ┆         ┆ cd…      │
+    └───────────┴──────────┴──────────┴──────────┴───┴────────┴─────────┴──────────┘
```

Comment:
### Note
Whenever I refer to `metadata.parquet`, this is the data I'm talking about.

Hoping this is a handy reference during discussions, since the file is a binary

https://github.com/vega/altair/blob/7ddb2a8c1e8ec6477cfc646c385e0b168f2fd330/altair/datasets/_metadata/metadata.parquet